### DISCUSSION TITLE
QCefView(QWidget * parent, Qt::WindowFlags f)构造函数的意义

### DISCUSSION THREAD

**Discussion by L-Super** [CefView/QCefView#260] [Q&A]
通常使用的是在构造的时候，就传入url：
```cpp
QCefView(const QString url, const QCefSetting * setting, QWidget * parent, Qt::WindowFlags f)
```

QCefView没有额外提供设置url的接口，那么`QCefView(QWidget * parent, Qt::WindowFlags f)` 构造函数的意义是什么？

**Comment by tishion** [comment]
```cpp
  /// <summary>
  /// Navigates to the URL
  /// </summary>
  /// <param name="url">The url</param>
  void navigateToUrl(const QString& url);
```

https://github.com/CefView/QCefView/blob/6d42213de75d528916f8b39d97ff711db7bd0586/include/QCefView.h#LL123C42-L123C42

**Reply by L-Super** [reply]
原来如此！感谢大佬

**Comment by L-Super** [comment]
我添加了以下构造函数，试图使用`QCefView(QWidget * parent, Qt::WindowFlags f)`进行构造
```cpp
//example/QCefViewTest/CefViewWidget.h
CefViewWidget(QWidget* parent = nullptr): QCefView(parent) {}
```
```cpp
//example/QCefViewTest/MainWindow.cpp
  m_pRightCefViewWidget = new CefViewWidget(this);
  m_pRightCefViewWidget->navigateToUrl("https://mirrors.tuna.tsinghua.edu.cn/qt/archive/online_installers/4.5/");
```
然后使用navigateToUrl()去加载网页，但是网页并未加载出来。经调试发现，经由`QCefViewPrivate::navigateToUrl(const QString& url)`这里的时候，pCefBrowser_为空指针，导致未能进入if语句里，导致无法加载url。
![image](https://github.com/CefView/QCefView/assets/40905056/7e523e22-7af7-45b4-950e-5fe4c6b78ccf)

不知道是我使用方式的问题，还是算一个bug

**Reply by L-Super** [reply]
@tishion

**Reply by tishion** [reply]
这是已知问题，因为底层创建CEF浏览器对象是异步操作，所以上层需要在确定底层浏览器创建好以后才能调用navigate这里可以优化一下。
